notes.lua

```diff
@@ -0,0 +1,322 @@
+end
+
+function NotesOverlay:onRenderFrame(dc)
+    if not df.global.pause_state and not dfhack.screen.inGraphicsMode() then
+        return
+    end
+
+    if self:viewportChanged() then
+        self:reloadVisibleNotes()
+    end
+
+    dc:map(true)
+
+    local texpos = dfhack.textures.getTexposByHandle(green_pin[1])
+    dc:pen({fg=COLOR_BLACK, bg=COLOR_LIGHTCYAN, tile=texpos})
+
+    for _, note in pairs(self.notes) do
+        dc
+            :seek(note.screen_pos.x, note.screen_pos.y)
+            :char('N')
+    end
+
+    dc:map(false)
+end
+
+function NotesOverlay:reloadVisibleNotes()
+    self.notes = {}
+
+    local viewport = guidm.Viewport.get()
+    self.viewport_pos = {
+        x=df.global.window_x,
+        y=df.global.window_y,
+        z=df.global.window_z
+    }
+
+    for _, point in ipairs(df.global.plotinfo.waypoints.points) do
+        if viewport:isVisible(point.pos) then
+            local pos = viewport:tileToScreen(point.pos)
+            table.insert(self.notes, {
+                point=point,
+                screen_pos=pos
+            })
+        end
+    end
+end
+
+NoteManager = defclass(NoteManager, gui.ZScreen)
+NoteManager.ATTRS{
+    focus_path='hotspot/menu',
+    hotspot_frame=DEFAULT_NIL,
+    note=DEFAULT_NIL,
+    on_update=DEFAULT_NIL,
+}
+
+function NoteManager:init()
+    local edit_mode = self.note ~= nil
+
+    self:addviews{
+        widgets.Window{
+            frame={w=35,h=20},
+            frame_inset={t=1},
+            autoarrange_subviews=false,
+            subviews={
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_N',
+                    label='Name',
+                    frame={t=0},
+                    on_activate=function() self.subviews.name:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='name',
+                    focus_path='notes/name',
+                    frame={t=1,h=3},
+                    frame_style=gui.FRAME_INTERIOR,
+                    frame_style_b=nil,
+                    init_text=self.note and self.note.name or ''
+                },
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_C',
+                    label='Comment',
+                    frame={t=4},
+                    on_activate=function() self.subviews.comment:setFocus(true) end,
+                },
+                text_editor.TextEditor{
+                    view_id='comment',
+                    frame={t=5,b=3},
+                    focus_path='notes/comment',
+                    frame_style=gui.FRAME_INTERIOR,
+                    init_text=self.note and self.note.comment or ''
+                },
+                widgets.Panel{
+                    view_id='buttons',
+                    frame={b=0,h=3},
+                    autoarrange_subviews=true,
+                    subviews={
+                        edit_mode and widgets.TextButton{
```

Comment:
For footer elements, we usually use `HotkeyLabel`. `TextButton` is intended for use in places where there are no other identifying markings that the widget is added by DFHack, such as in overlay panels that don't have room for a branded frame (e.g. the search widget on the Tasks panel).